### ISSUE TITLE
[release 0.7.1]: Plugging in CA certificate and key failed on Verify CA certificate

### ISSUE THREAD

**Issue by janevalleye2e** [istio/istio#4925]
Ref doc: https://istio.io/docs/tasks/security/plugin-ca-cert.html
1. ran through the steps following the https://istio.io/docs/tasks/security/plugin-ca-cert.html
2. on Verifying the new certificates, step 4
```
tail /tmp/pod-cert-chain.pem -n 22 > /tmp/pod-cert-chain-ca.pem
tail: -n: No such file or directory
tail: 22: No such file or directory
```
modified and ran following commands
```
tail -n 22 /tmp/pod-cert-chain.pem  > /tmp/pod-cert-chain-ca.pem
openssl x509 -in /tmp/ca-cert.pem -text -noout > /tmp/ca-cert.crt.txt
openssl x509 -in /tmp/pod-cert-chain-ca.pem -text -noout > /tmp/pod-cert-chain-ca.crt.txt
diff /tmp/ca-cert.crt.txt /tmp/pod-cert-chain-ca.crt.txt
```
**Return differences, Expect that the output to be empty**
```
diff /tmp/ca-cert.crt.txt /tmp/pod-cert-chain-ca.crt.txt
5c5
<             e3:75:89:fa:c1:67:6f:c1
---
>             9e:5e:83:37:72:3d:43:c4:2f:3e:34:b5:91:a7:77:c4
7c7
<         Issuer: C=US, ST=California, L=Sunnyvale, O=Istio, OU=Test, CN=Root CA/emailAddress=testrootca@istio.io
---
>         Issuer: C=US, ST=California, L=Sunnyvale, O=Istio, CN=Istio CA
9,11c9,11
<             Not Before: Jan 24 19:15:51 2018 GMT
<             Not After : Dec 31 19:15:51 2117 GMT
<         Subject: C=US, ST=California, L=Sunnyvale, O=Istio, CN=Istio CA
---
>             Not Before: Apr 12 23:10:41 2018 GMT
>             Not After : Apr 13 18:10:41 2018 GMT
>         Subject: O=
16,33c16,33
<                     00:cb:30:b1:af:fc:6e:d3:3c:b9:ad:50:62:b2:8f:
<                     5f:7e:09:74:d1:b4:4a:bc:1f:c7:17:80:17:f7:fc:
<                     ad:99:9e:87:aa:cc:b5:dd:72:10:93:cf:ee:fc:1c:
<                     bd:88:2b:d5:c1:72:0c:bf:24:f4:09:b8:89:ab:f6:
<                     8f:12:3e:66:25:4c:b4:97:36:eb:52:cd:77:a2:77:
<                     97:aa:b3:d7:7f:b8:ab:dc:7c:dd:47:0f:92:06:15:
<                     e5:b2:1f:73:00:f6:49:5d:c1:7d:dc:32:54:dc:66:
<                     9b:3c:ac:01:bc:62:74:21:53:09:3c:81:42:b8:32:
<                     0f:c1:6e:24:14:02:3b:47:ff:00:e4:b4:9d:3e:b1:
<                     71:e8:4c:8c:5e:5e:ca:33:c8:de:90:2d:32:f4:39:
<                     d3:8f:cd:ff:7d:8e:f6:59:c5:97:ed:84:e9:81:97:
<                     81:1c:07:9e:40:e3:d3:67:62:98:6c:56:a5:da:02:
<                     ec:6a:be:bd:3e:01:52:d1:3a:26:11:23:bd:33:5e:
<                     18:8a:de:e6:cc:1d:56:0c:ad:b3:f6:0d:eb:fb:32:
<                     f1:10:d7:49:e4:91:bf:66:c9:0a:7b:e4:ce:e0:38:
<                     aa:8b:93:89:b7:f8:7c:ca:ca:52:d5:c9:3c:2c:9b:
<                     42:a2:57:bd:f7:5f:6e:98:1c:a0:e6:8a:ee:7e:5a:
<                     88:95
---
>                     00:c8:e2:b4:88:df:7d:bb:7e:5b:25:78:18:94:5b:
>                     e0:92:0a:b0:0d:98:fa:70:0f:93:15:f6:e1:6a:94:
>                     e4:83:0f:a7:13:1d:03:58:b1:55:dd:17:49:45:73:
>                     b2:ce:76:13:8c:c0:d7:fc:fa:b8:b5:93:9e:a8:f2:
>                     5c:41:b0:76:50:92:06:11:f6:3d:1c:45:52:78:12:
>                     6c:71:c6:a9:34:e0:63:4a:36:50:18:11:13:cc:63:
>                     c7:a7:f1:83:22:1f:20:6e:93:20:8d:d1:71:3d:79:
>                     a8:25:cf:60:9e:50:ae:db:d3:35:52:8d:04:8b:2c:
>                     12:01:48:1d:20:d5:77:62:6d:7b:92:c0:a3:65:1b:
>                     4d:bc:2c:6e:aa:3c:2f:49:38:b3:75:c6:08:8d:d3:
>                     38:bc:73:ed:8d:2e:3d:5f:fe:60:30:ca:cc:7c:b6:
>                     6d:20:45:5e:e3:c4:e6:86:46:57:ae:d1:ba:dc:a8:
>                     c5:7b:d7:ea:5e:5d:16:c5:5c:87:be:b7:92:df:c1:
>                     30:d3:de:23:da:c9:1d:e1:1b:09:d1:c6:37:c3:02:
>                     ed:7d:ff:56:d9:8b:95:c5:42:d1:4d:ba:d6:55:0e:
>                     43:c8:f5:9e:ba:46:b0:9c:9c:03:ee:f4:2b:31:c1:
>                     cf:7d:9c:1c:5d:21:65:78:34:aa:f8:10:05:9c:c9:
>                     5b:fb
36,39c36,41
<             X509v3 Key Usage:
<                 Certificate Sign
<             X509v3 Basic Constraints:
<                 CA:TRUE
---
>             X509v3 Key Usage: critical
>                 Digital Signature, Key Encipherment
>             X509v3 Extended Key Usage:
>                 TLS Web Server Authentication, TLS Web Client Authentication
>             X509v3 Basic Constraints: critical
>                 CA:FALSE
41c43
<                 DNS:ca.istio.io
---
>                 URI:spiffe://cluster.local/ns/default/sa/default
43,57c45,59
<          96:d1:c4:86:1c:80:b2:f7:b8:2b:86:cb:80:15:ed:1f:05:b3:
<          a3:a4:a9:17:37:40:7d:7a:4b:4a:1a:4e:24:db:50:34:44:5b:
<          de:a8:3a:89:41:90:75:8f:b0:03:60:69:99:29:0a:af:74:70:
<          ff:a0:a5:3f:9f:9c:08:6e:df:1a:87:0f:69:98:7e:73:ee:0d:
<          c6:6f:6c:b8:cb:99:05:90:ad:34:e7:98:38:c2:1f:bc:23:dc:
<          09:2f:bc:3d:0f:f4:84:2e:3f:d0:9d:d6:24:ac:c6:06:cb:1a:
<          68:cb:9f:27:1c:e0:14:f3:ed:0c:9f:7d:f2:46:3c:ae:54:4e:
<          82:17:52:69:5b:78:db:7f:e3:5c:80:18:9c:fd:d8:d3:99:fc:
<          4c:61:11:06:3e:d3:2d:db:48:55:bb:54:23:1d:c9:1c:0a:37:
<          3d:62:72:c0:0e:a6:10:3a:2f:14:c2:61:24:44:69:f6:c0:c7:
<          22:bd:41:6e:15:ca:d5:31:4d:67:a1:cd:1f:a8:08:9e:44:92:
<          32:db:c2:e0:fb:5e:8d:be:f8:f4:f4:da:78:f4:4f:17:b2:03:
<          ff:f7:aa:ff:70:07:f1:42:b9:d0:5d:d3:41:c2:e3:12:a5:94:
<          b0:95:27:68:52:a5:b4:e7:3e:44:08:f3:0e:49:5c:50:f7:86:
<          e1:8e:9e:30
---
>          14:7d:21:2d:6c:8b:66:3a:1a:c1:bf:6b:98:10:3e:28:d1:08:
>          1f:d1:72:5b:8f:ab:19:e4:e7:c2:06:df:7e:02:67:4e:c8:e1:
>          c6:12:ec:7d:16:ea:a4:b1:bc:14:0a:e9:2d:b4:29:62:c5:62:
>          93:4d:77:21:ee:63:b7:47:5c:94:03:52:d8:fc:58:16:f9:ef:
>          5d:b4:e6:ed:5b:3b:43:aa:eb:19:50:8b:ee:9d:3d:fb:89:48:
>          55:62:04:40:16:62:f2:5d:20:ea:de:fc:bd:10:ac:0c:de:f0:
>          34:fc:ca:c9:0f:45:23:27:4e:c3:d4:f4:25:b3:5b:d5:a1:2c:
>          b4:05:db:3f:8d:5c:2a:6e:08:18:c8:c9:bd:a1:ae:92:38:17:
>          68:c2:96:6b:f3:53:04:84:bf:12:0f:4c:c1:6f:50:3e:62:43:
>          ec:04:35:44:95:44:ab:04:48:4c:3c:5c:61:60:80:57:dd:fd:
>          35:90:28:5e:a4:fa:5a:90:a2:53:ba:52:d9:6e:b8:dc:85:20:
>          36:fc:e9:c3:c2:f9:d8:0b:2d:3c:e0:2e:0d:d4:2f:06:5a:1f:
>          ba:63:e8:19:92:9b:04:38:49:30:11:e2:ec:b8:ff:b5:88:a2:
>          30:c9:cb:35:d9:34:63:33:c2:a7:d7:22:5a:51:8a:25:f3:c5:
>          e9:6a:38:49
```

**Comment by janevalleye2e** [comment]
@kidiyoor

**Comment by janevalleye2e** [comment]
@liminw

**Comment by janevalleye2e** [comment]
@vilobhmm

**Comment by myidpt** [comment]
Sorry this is a known issue on release V0.7.1. It is fixed by:
https://github.com/istio/istio/commit/3a4bad74995d7c0e6b33bd3ff82d2dfb18f4ab71
As a work around, you can use V0.6.0 Istio CA. Otherwise please wait for V0.8 to come out.

**Comment by kidiyoor** [comment]
thanks  @myidpt. How different is 0.7.1 CA compared to 0.6.0.
Will rest of 0.7.1 istio components be able to work with 0.6.0 CA ?

**Comment by myidpt** [comment]
Yes, the CA is pretty stand-alone from other components. It should work with other V0.7.1 components.

**Comment by bhaskar-chenchu** [comment]
Hope this issue is get fixed in the Istio 0.8.0 version? Please confirm
